### PR TITLE
Add video inference

### PR DESCRIPTION
**Pull request by sokovninn**
This pull request includes several changes to enhance the inference capabilities of the `luxonis_train` module, particularly by adding support for video files and improving the handling of image and video sources. The most important changes include renaming variables for clarity, adding support for video processing, and updating the inference logic to accommodate these improvements.

### Enhancements to Inference Capabilities:

* [`luxonis_train/__main__.py`](diffhunk://#diff-edffd69fae1e1a3d23de321080cb35407f9591a2e03e22ba74074d48f9edec29L53-R56): Renamed `ImgPathType` to `SourcePathType` and updated the help text to include video files. Updated the `infer` function to use `source_path` instead of `img_path`. 

* [`luxonis_train/core/core.py`](diffhunk://#diff-aaa0e36bee1c6c3cd663a8dd04a39a455a1f4f21261356a5e86fc4bfc938bdb1R40-R59): Added `process_video` to the imports. Defined `IMAGE_FORMATS` and `VIDEO_FORMATS` sets for supported file types. Updated the `infer` method to handle video files and directories containing images.

### Improvements in Utility Functions:

* [`luxonis_train/core/utils/infer_utils.py`](diffhunk://#diff-c1fe3d7b3a9d1b7367a23592cfa87b8962cb754fe295823e6662c81611bf404eR1-R24): Added `process_video` function to handle video file inference. Refactored `process_images` and introduced `prepare_and_infer_image` for better code reuse. Updated `render_visualizations` to optionally return visualizations instead of displaying them.

### Possible improvements:

- Parallel video processing and batched inference
- Webcam/stream source for inference

### CODE REVIEW

File: luxonis_train/core/core.py [hunk 1]
```diff
@@ -37,11 +37,26 @@
 from .utils.infer_utils import (
     process_dataset_images,
     process_images,
+    process_video,
 )
 from .utils.train_utils import create_trainer
 
 logger = getLogger(__name__)
 
+IMAGE_FORMATS = {
```
Comment (klemen1999) [inline luxonis_train/core/core.py:46]: Let's move these two sets into utils/infer_utils.py

File: luxonis_train/core/utils/infer_utils.py [hunk 2]
```diff
@@ -27,13 +33,93 @@ def render_visualizations(
...
+    if save_dir is None and show:
         if cv2.waitKey(0) == ord("q"):
             exit()
 
+    return rendered_visualizations
+
+
+def prepare_and_infer_image(model, img: np.ndarray, labels: dict, view: str):
+    """Prepares the image for inference and runs the model."""
+    img = cv2.cvtColor(img, cv2.COLOR_BGR2RGB)
```
Comment (klemen1999) [inline luxonis_train/core/utils/infer_utils.py:50]: This part actually depends if `trainer.preprocessing.train_rgb` is set to True or False in the config (True by default - means that we should feed RGB in the model).

**Review by klemen1999 — APPROVED** [review]
LGTM, left some comments

File: luxonis_train/core/utils/infer_utils.py (hunk 2 above)
Comment (sokovninn) [reply luxonis_train/core/utils/infer_utils.py:50]: If I am not mistaken, this should be handled in the augmentations which take RGB images.

Comment (klemen1999) [reply luxonis_train/core/utils/infer_utils.py:64]: Oh you, you are right.

**Review by kozlov721 — APPROVED** [review]
LGTM

**Pull request merged**